modmesh/app/euler1d.py

```diff
@@ -442,288 +453,266 @@ def _update_single_figure(self):
                     ax_new.yaxis.set_major_formatter((FormatStrFormatter
                                                       ('%.2f')))
                 select_num += 1
-            lines.append(data.ana)
-            lines.append(data.num)
 
         ax.set_xlabel("distance (m)")
         ax.grid()
-        canvas.installEventFilter(self)
 
         # These parameters are the results obtained by my tuning on GUI
         fig.subplots_adjust(left=0.1,
                             right=0.97 - (self.checkbox_select_num - 1) * 0.1,
                             bottom=0.093, top=0.976,
                             wspace=0.2, hspace=0.2)
 
-        prev_fig = self.figure_layout.takeAt(0)
-
-        if prev_fig:
-            prev_fig.widget().deleteLater()
-
         self.update_lines()
-        self.figure_layout.addWidget(canvas)
-
-        return lines
 
-    def _build_checkbox(self, layout):
-        layout.setSpacing(50)
-        for name, data_line in self.data_lines.items():
-            check_box = QCheckBox(name)
-            if data_line[1]:
-                check_box.toggle()
-            check_box.clicked.connect(self._checkbox_cb)
-            layout.addWidget(check_box)
+        return canvas
 
-    @Slot(bool)
-    def _checkbox_cb(self, checked=False):
+    def march_alpha2(self, steps=1):
         """
-        This callback function don't care button's checked state,
-        therefore the checked state is not used in this function.
+        This function is used to call c++ solver to march the time step, also
+        calling python side analytical solution to march the time step.
 
-        Under a single plot layout, that allow 3 lines on the same chart
-        simultaneously to avoid it looking too crowded.
-        I have chosen to use checkboxes for user to select
-        the parameters they want to plot on the chart.
-
-        :param checked: button is checked or not
-        :return: nothing
+        :return: None
         """
-        checkbox = self.sender()
-        if self.checkbox_select_num == 3:
-            if checkbox.isChecked():
-                checkbox.toggle()
-            else:
-                self.checkbox_select_num -= 1
-                self.data_lines[checkbox.text()][1] = False
-        elif self.checkbox_select_num == 1:
-            if not checkbox.isChecked():
-                checkbox.toggle()
-            else:
-                self.checkbox_select_num += 1
-                self.data_lines[checkbox.text()][1] = True
-        else:
-            if checkbox.isChecked():
-                self.checkbox_select_num += 1
-                self.data_lines[checkbox.text()][1] = True
-            else:
-                self.checkbox_select_num -= 1
-                self.data_lines[checkbox.text()][1] = False
-
-        self._update_single_figure()
-
+        if self.max_steps and self.current_step > self.max_steps:
+            self.stop()
+            return
 
-class Controller:
```

Comment:
The `Controller` class was removed because it can be replaced by a configuration windows, that allows user modified the solver configuration via GUI.